File: 2025-10/2025-10-15/htmlTagStripper.js
function stripTags(html) {

  return html.replace(/<[^>]+>/g, "");
}

console.log(stripTags('<a href="#">Click here</a>'));
console.log(stripTags('<p class="center">Hello <b>World</b>!</p>'));
console.log(stripTags('<img src="cat.jpg" alt="Cat">'));
console.log(stripTags('<main id="main"><section class="section">section</section><section class="section">section</section></main>'));

// --- TEST SUITE ---

const testsText = `
1. stripTags('<a href="#">Click here</a>') should return "Click here".
2. stripTags('<p class="center">Hello <b>World</b>!</p>') should return "Hello World!".
3. stripTags('<img src="cat.jpg" alt="Cat">') should return "".
4. stripTags('<main id="main"><section class="section">section</section><section class="section">section</section></main>') should return "sectionsection".
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+)\.$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let allPassed = true;
  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);
    let comparison;

    if (Array.isArray(testOutput))
      comparison = arraysEqual(functionCallOutput, testOutput);
    else
      comparison = functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      allPassed = false;
      failCount ++;
    }
  })

  console.log("----------------------------");

  if (allPassed)
    console.log("🎉SUCCESS: All tests PASSED.");
  else
    console.log(`⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`);
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (const index in a)
    if (a[index] !== b[index])
      return false;
  
  return true;
}

runTests(testData);

File: 10 October/10-15/html_tag_stripper.py
def strip_tags(html):
    import re

    return re.sub("<[^>]+>", "", html)

print(strip_tags('<a href="#">Click here</a>'))
print(strip_tags('<p class="center">Hello <b>World</b>!</p>'))
print(strip_tags('<img src="cat.jpg" alt="Cat">'))
print(strip_tags('<main id="main"><section class="section">section</section><section class="section">section</section></main>'))

# --- TEST SUITE ---

tests_text = '''
1. strip_tags('<a href="#">Click here</a>') should return "Click here".
2. strip_tags('<p class="center">Hello <b>World</b>!</p>') should return "Hello World!".
3. strip_tags('<img src="cat.jpg" alt="Cat">') should return "".
4. strip_tags('<main id="main"><section class="section">section</section><section class="section">section</section></main>') should return "sectionsection".
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+)\.$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    all_passed = True
    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            all_passed = False
            fail_count += 1

    print("----------------------------");

    if (all_passed):
        print("🎉 SUCCESS: All tests PASSED.");
    else:
        print(f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED.");

run_tests(test_data)